tests/data_ingestion/test_group_timeseries_resampler.py

```diff
@@ -0,0 +1,77 @@
+"""
+Tests for the `ComponentMetricGroupResampler`
+
+Copyright
+Copyright © 2021 Frequenz Energy-as-a-Service GmbH
+
+License
+MIT
+"""
+from datetime import datetime, timedelta
+from typing import Sequence
+
+import pytz
+
+from frequenz.sdk.data_ingestion.resampling.timeseries_group_resampler import (
+    ComponentMetricGroupResampler,
+)
+from frequenz.sdk.data_ingestion.resampling.timeseries_resampler import Sample
+
+
+# pylint: disable=unused-argument
+def resampling_function_sum(
+    samples: Sequence[Sample], resampling_period_s: float
+) -> float:
+    """Calculate sum of the provided values.
+
+    Args:
+        samples: sequences of samples to apply the average to
+        resampling_period_s: value describing how often resampling should be performed,
+            in seconds
+
+    Returns:
+        sum of all the sample values
+    """
+    if len(samples) == 0:
+        return 0.0
+    return sum(sample.value for sample in samples if sample.value is not None)
+
+
+def test_component_metric_group_resampler() -> None:
+    """Test if resampling is properly delegated to component metric resamplers."""
+    resampler = ComponentMetricGroupResampler(
+        resampling_period_s=0.2,
+        max_data_age_in_periods=5.0,
+        initial_resampling_function=resampling_function_sum,
+    )
+
+    time_series_id_1 = "123_active_power"
+    time_series_id_2 = "99_active_power"
+
+    resampler.add_time_series(time_series_id=time_series_id_1)
+    resampler.add_time_series(time_series_id=time_series_id_2)
+
+    timestamp = datetime.now(tz=pytz.UTC)
```

Comment:
I think it would be a good idea to mock `datetime.now`, so it is deterministic and we avoid the risk of flaky tests. For example if for some reason here it would take 0.1s for the test to reach the `resample()` call, the test will break (because one sample's timestamp is now-0.9s).

Here you can do it by patching `frequenz.sdk.data_ingestion.resampling.datetime`.

We could consider using [`time-machine`](https://pypi.org/project/time-machine/), as I think it will be useful to mock time in many different tests.

Unrelated side note: I first found `freezegun`, which seems to be the most popular choice, but it looks like `time-machine` is more modern and came as a [response to `freezegun` very poor performance](https://adamj.eu/tech/2021/02/19/freezegun-versus-time-machine/).

Reply:
I can mock it if needed, but actually I wouldn't say it is necessary in this case (note that there are no `async` calls here) so it is quite easy to set up the test conditions with plain timestamps.

Also, we already have a lot of tests that rely on timestamps set up like this so if I mock it here, we might want to mock it in other places too?

I've used the `freezegun` and it did the job. I am not sure why performance is such a big deal here since it is only for testing purposes and we don't have any tests executing thousands or millions of iterations ;) 

It is up to you, please let me if I should mock `datetime.now` and using which lib.

Reply:
> I can mock it if needed, but actually I wouldn't say it is necessary in this case (note that there are no async calls here) so it is quite easy to set up the test conditions with plain timestamps.

It is not related to async, it is related to the simple pass of time, which could lead to flakyness.

> Also, we already have a lot of tests that rely on timestamps set up like this so if I mock it here, we might want to mock it in other places too?

Yes, eventually.

> I've used the freezegun and it did the job. I am not sure why performance is such a big deal here since it is only for testing purposes and we don't have any tests executing thousands or millions of iterations ;)

Well, it turns out the performance penalty comes from the amount of modules being imported, not the times you use the decorator. The post is an interesting read and quite short. I agree that this will probably not be a problem for us for now, but I also don't see any reason to use a library that has performance problems if there is a better alternative that does exactly the same...

> It is up to you, please let me if I should mock datetime.now and using which lib.

Unless @ela-kotulska-frequenz or @sahas-subramanian-frequenz have anything against this, I say do it and use `time-machine`. Make sure the dependency is only added for testing.

Reply:
![image](https://user-images.githubusercontent.com/108664423/198557477-b5685606-c01c-4ac6-8c30-3598e0e244e3.png)